docs/testing/overview.md

```diff
@@ -1,6 +1,52 @@
 # Testing Overview
 
-At MetaMask, we are committed to delivering high-quality software. A key aspect of achieving this goal is through a robust testing strategy that includes both end-to-end (e2e), integration and unit tests. This document provides comprehensive guidelines on each testing type, emphasizing the importance of choosing the right test for the right scenario and detailing the specific practices for writing and organizing these tests.
+At MetaMask, we are committed to delivering high-quality software. These guidelines describe how we use automated tests to maximize product quality while minimizing the cost of writing and maintaining tests.
+
+Our automated tests fall into three general categories: end-to-end (e2e), integration and unit tests. This document provides comprehensive guidelines on each testing type, emphasizing the importance of choosing the right test for the right scenario and detailing the specific practices for writing and organizing these tests.
+
+## Testing principles
+
+What makes a good test? Here are some general principles.
+
+### 1. Tests should describe expectations
+
+What's the difference between a bug and a feature? A feature is expected, a bug is not.
+
+If it's not clear what the code-under-test is supposed to do, it won't be clear what to do when the test fails either. Clear expectations are essential for an effective test.
+
+### 2. Tests should catch bugs
+
+This is the main purpose of tests. Take care to ensure that each test actually fails when the described expectation is not met.
+
+### 3. Tests should not break often
+
+It's expected that tests will break when we change the behavior or APIs under test. But if a test breaks often from refactors, or from changes to behavior or APIs not directly under test, it's sign that the test may be poorly designed.
+
+Test breakages are expensive. We should do what we can to minimize them.
+
+### 4. Tests should be easy to read
+
+A test isn't effective if we can't understand why it failed. The more readable a test is, the faster we can fix it.
+
+### 5. Tests should be easy to write
+
+Some behaviors are inherently more complex than others, increasing the difficulty of testing them. But in general we want tests to be easy to write, so that we can keep costs low.
+
+If a test is hard to write, consider that as a warning that you might be approaching the problem in a non-optimal way. Resist the urge to power through the work, and instead consider how redesigning the code-under-test or using a different testing strategy might save time.
```

Comment:
In these cases, I'd recommend protecting against major regressions using end-to-end tests and/or integration tests. If the test is sufficiently distant from the code, it will be easier to refactor/redesign. This can give you some safety while you work to make the code easier to test comprehensively.

I considered adding guidance like this here, but it felt a bit out of place. I was trying to keep these principles minimal, with specific guidance elsewhere. I am not sure where this advice would best fit though 🤔

Reply:
Maybe I can include this in the last principle, where it talks about deleting tests / not writing bad tests. I can have a sentence at the end about this.

Reply:
I wasn't able to find a place for this. I'll look elsewhere for where this advice can live, in a later PR.